jax/experimental/jax2tf/tfjs_demo/Sketcher/sketcher.py

```diff
@@ -0,0 +1,278 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from absl import app # type: ignore
+import os # type: ignore
+import time
+from typing import Any, Callable, List, Sequence
+
+import flax # type: ignore
+from flax import linen as nn
+from flax.training import common_utils # type: ignore
+
+import jax # type: ignore
+from jax import lax
+from jax import numpy as jnp
+import numpy as np # type: ignore
+import tensorflow as tf # type: ignore
+from tensorflowjs.converters import convert_tf_saved_model # type: ignore
+
+from jax.config import config # type: ignore
+config.config_with_absl()
+
+from jax.experimental import jax2tf # type: ignore
+jax2tf.jax2tf.ENABLE_TF_CONVOLUTION = True
+
+TfVariable = Any
+
+class ExportWrapper(tf.train.Checkpoint):
+
+  def __init__(self, fn: Callable, params,
+               input_shape: Sequence[int],
+               batch_sizes: Sequence[int]):
+    """
+    Args:
+      fn: a function taking two arguments, the parameters and the batch of
+        images.
+      params: the parameters, as a list/tuple/dictionary of np.ndarray, to be
+        used as first argument for `fn`.
+     input_shape: the shape of the second argument of `fn` (except the batch size)
+     batch_sizes: a sequence of batch sizes for which to save the function.
+    """
+    super().__init__()
+    # Convert fn from JAX to TF.
+    self._fn = jax2tf.convert(fn, with_gradient=True)
```

Comment:
Was it important to add `with_gradient`?

Reply:
Yes, as `PreventGradient` is not a valid op for the conversion!